CMakeLists.txt

```diff
@@ -256,55 +256,55 @@ if(WIN32)
 endif()
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
 
-# Get the current working branch
-execute_process(
-  COMMAND git rev-parse --abbrev-ref HEAD
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GIT_BRANCH
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  ERROR_QUIET
-)
-if(NOT GIT_BRANCH)
-  set(GIT_BRANCH "unknown")
-endif()
-message(STATUS "Git branch: ${GIT_BRANCH}")
-
-# Get the latest abbreviated commit hash of the working branch
-execute_process(
-  COMMAND git log -1 --format=%h
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GIT_COMMIT_HASH
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  ERROR_QUIET
-)
-if(NOT GIT_COMMIT_HASH)
-    set(GIT_COMMIT_HASH "unknown")
-endif()
-message(STATUS "Git commit: ${GIT_COMMIT_HASH}")
-
-# Get the number of commits on the working branch
-execute_process(
-  COMMAND git rev-list --count --first-parent HEAD
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GIT_COMMIT_COUNT
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  ERROR_QUIET
-)
-if(NOT GIT_COMMIT_COUNT)
-    set(GIT_COMMIT_COUNT "unknown")
+if(NOT GIT_TAG)
+  execute_process(
+    COMMAND git describe --tags --abbrev=0
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_TAG
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+  )
 endif()
+message(STATUS "Git tag: ${GIT_TAG}")
 
-# Check if the worktree is dirty
-execute_process(
-  COMMAND git diff --quiet
-  RESULT_VARIABLE GIT_WORKTREE_DIRTY
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  ERROR_QUIET
-)
-if(GIT_WORKTREE_DIRTY EQUAL "1")
-  set(GIT_COMMIT_COUNT "${GIT_COMMIT_COUNT}+")
+if(NOT GIT_COMMIT_DESCRIPTION)
+  execute_process(
+    COMMAND git describe --tags --always --dirty=-modified
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_COMMIT_DESCRIPTION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+  )
 endif()
-message(STATUS "Git commit count: ${GIT_COMMIT_COUNT}")
+message(STATUS "Git commit: ${GIT_COMMIT_DESCRIPTION}")
+
+if(NOT GIT_COMMIT_DATE)
+  set(ENV{TZ} UTC)
+  execute_process(
+    COMMAND git show --quiet --format=%cd --date=short
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_COMMIT_DATE
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+  )
+endif()
+message(STATUS "Git commit date: ${GIT_COMMIT_DATE}")
+
+string(REGEX MATCH "([0-9]*)-" GIT_YEAR "${GIT_COMMIT_DATE}")
+
+if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/git_info.h")
+  if(NOT GIT_TAG OR NOT GIT_COMMIT_DESCRIPTION OR NOT GIT_COMMIT_DATE)
+    message(FATAL_ERROR "Version information not found. "
+    "Either clone the Git repository from https://github.com/mixxxdj/mixxx.git ; "
+    "download an official release tarball from https://downloads.mixxx.org ; "
+    "or manually specify GIT_TAG, GIT_COMMIT_DESCRIPTION, and GIT_COMMIT_DATE variables to CMake.")
+  endif()
+  configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/git_info.h.template"
```

Comment:
> please add a comment which variables are used in git_info.h

I intentionally removed that because it is one more thing that can become out of date. IMO it is best to look at the template file to see what it does.

Reply:
Please re-add it. we need it as find hock if one verifies if a variable is still in use or not.

Reply:
That is IMO just as good a reason to not add such a comment. It will easily become incorrect and therefore it would be misleading to judge if a variable is still used by the comment.

Reply:
This bites me already. It is a good practice to do that. So please do.

Reply:
And it has bitten me already that the comment became wrong after making changes.

Reply:
The code is read way more often than changed. The comment as find hock as been proved as useful. 

We have no other chance here, because configure_file has no option to specify the replaced vars. 
They are out of sight here, so please add.